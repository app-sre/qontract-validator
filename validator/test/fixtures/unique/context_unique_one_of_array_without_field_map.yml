schemas:
  "schema-1.yml":
    properties:
      one_of_array:
        type: array
        items:
          oneOf:
          - additionalProperties: false
            properties:
              simple_field:
                type: string
              a_field:
                type: string
          - additionalProperties: false
            properties:
              simple_field:
                type: string
              b_field:
                type: string
graphql:
  $schema: "graphql-schema-1.yml"
  confs:
  - name: Schema_v1
    fields:
    - name: one_of_array
      type: OneOfSchema_v1
      isList: true
  - name: OneOfSchema_v1
    fields:
    - name: simple_field
      type: string
      isContextUnique: true
    - name: a_field
      type: string
    - name: b_field
      type: string
  - name: Query
    fields:
    - type: Schema_v1
      name: schema_v1
      datafileSchema: "schema-1.yml"
data:
  file-1-schema-1.yml:
    $schema: schema-1.yml
    one_of_array:
    - simple_field: bla
      a_field: a
    - simple_field: bla
      b_field: b
resources: {}
