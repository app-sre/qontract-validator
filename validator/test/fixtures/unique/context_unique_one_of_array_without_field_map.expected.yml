schemas:
  "schema-1.yml":
    properties:
      one_of_array:
        type: array
        items:
          oneOf:
          - additionalProperties: false
            properties:
              simple_field:
                type: string
              a_field:
                type: string
              __identifier: # +
                type: string
          - additionalProperties: false
            properties:
              simple_field:
                type: string
              b_field:
                type: string
              __identifier: # +
                type: string
graphql:
  $schema: "graphql-schema-1.yml"
  confs:
  - name: Schema_v1
    fields:
    - name: one_of_array
      type: OneOfSchema_v1
      isList: true
  - name: OneOfSchema_v1
    fields:
    - name: simple_field
      type: string
      isContextUnique: true
    - name: a_field
      type: string
    - name: b_field
      type: string
  - name: Query
    fields:
    - type: Schema_v1
      name: schema_v1
      datafileSchema: "schema-1.yml"
data:
  file-1-schema-1.yml:
    $schema: schema-1.yml
    one_of_array:
    - simple_field: bla
      a_field: a
      __identifier: "5fcc28b49951ac3dd925eacdbca8bee03d70cf686aa7d672a32545a15c0be787" # +
    - simple_field: bla
      b_field: b
      __identifier: "5fcc28b49951ac3dd925eacdbca8bee03d70cf686aa7d672a32545a15c0be787" # +
resources: {}
